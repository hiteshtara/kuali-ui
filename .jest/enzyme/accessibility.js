/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
