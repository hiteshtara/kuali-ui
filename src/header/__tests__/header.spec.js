/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import Header from '..'

describe('Header Component', () => {
  it('should render a Header component with correct properties', () => {
    const wrapper = shallow(
      <Header
        id='Header'
        menuItems={[
          {
            primaryText: 'User Settings',
            onClick: () => {}
          }
        ]}
        module={{ name: 'Conflict of Interest' }}
        onBrandSelected={() => {}}
        user={{ name: 'Slartibartfast' }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a Header component with no user or module name', () => {
    const wrapper = shallow(
      <Header id='Header' module={{ name: '' }} onBrandSelected={() => {}} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a Header with a custom logo', () => {
    const wrapper = shallow(
      <Header
        id='Header'
        module={{ name: 'COI' }}
        onBrandSelected={() => {}}
        Logo={<img src='https://localhost/logo.png' />}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
