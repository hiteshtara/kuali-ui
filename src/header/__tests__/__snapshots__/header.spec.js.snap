// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render a Header component with correct properties 1`] = `
<header
  className="layout__header"
>
  <span
    className="header__title"
  >
    Conflict of Interest
  </span>
  <span
    className="header__brand-logo-container"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="link"
    tabIndex={0}
  >
    <BrandLogo
      className="header__brand-logo"
    />
  </span>
  <div
    className="header__brand-divider"
  />
  <Menu
    anchor={
      Object {
        "x": "inner right",
        "y": "overlap",
      }
    }
    cascadingAnchor={
      Object {
        "x": "right",
        "y": "overlap",
      }
    }
    cascadingZDepth={3}
    className="header__menu-container"
    fixedTo={[Window]}
    id="header-menu"
    listHeightRestricted={true}
    listStyle={
      Object {
        "right": 10,
        "top": "100%",
      }
    }
    listZDepth={2}
    onClick={[Function]}
    onClose={[Function]}
    onKeyPress={[Function]}
    position="tr"
    repositionOnResize={false}
    repositionOnScroll={true}
    sameWidth={false}
    simplified={true}
    toggle={
      <AccessibleFakeButton
        className="header__menu-button"
        component="div"
        listenToEnter={true}
        listenToSpace={true}
        noFocusOutline={true}
        role="button"
        tabIndex={0}
      >
        <withInk(UserDetails)
          inkTransitionEnterTimeout={450}
          inkTransitionLeaveTimeout={300}
          inkTransitionOverlap={150}
          user={
            Object {
              "name": "Slartibartfast",
            }
          }
        />
      </AccessibleFakeButton>
    }
    visible={false}
  >
    <ListItem
      activeClassName="md-text--theme-primary"
      animateNestedItems={true}
      component="div"
      expanderIcon={
        <FontIcon
          iconClassName="material-icons"
        >
          keyboard_arrow_down
        </FontIcon>
      }
      itemComponent="li"
      key="0"
      onClick={[Function]}
      primaryText="User Settings"
      renderChildrenOutside={false}
    />
  </Menu>
</header>
`;

exports[`Header Component should render a Header component with no user or module name 1`] = `
<header
  className="layout__header"
>
  <span
    className="header__title"
  />
  <span
    className="header__brand-logo-container"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="link"
    tabIndex={0}
  >
    <BrandLogo
      className="header__brand-logo"
    />
  </span>
  <Menu
    anchor={
      Object {
        "x": "inner right",
        "y": "overlap",
      }
    }
    cascadingAnchor={
      Object {
        "x": "right",
        "y": "overlap",
      }
    }
    cascadingZDepth={3}
    className="header__menu-container"
    fixedTo={[Window]}
    id="header-menu"
    listHeightRestricted={true}
    listStyle={
      Object {
        "right": 10,
        "top": "100%",
      }
    }
    listZDepth={2}
    onClick={[Function]}
    onClose={[Function]}
    onKeyPress={[Function]}
    position="tr"
    repositionOnResize={false}
    repositionOnScroll={true}
    sameWidth={false}
    simplified={true}
    toggle={
      <AccessibleFakeButton
        className="header__menu-button"
        component="div"
        listenToEnter={true}
        listenToSpace={true}
        noFocusOutline={true}
        role="button"
        tabIndex={0}
      />
    }
    visible={false}
  />
</header>
`;

exports[`Header Component should render a Header with a custom logo 1`] = `
<header
  className="layout__header"
>
  <span
    className="header__title"
  >
    COI
  </span>
  <span
    className="header__brand-logo-container"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="link"
    tabIndex={0}
  >
    <img
      className="header__brand-logo"
      src="https://localhost/logo.png"
    />
  </span>
  <Menu
    anchor={
      Object {
        "x": "inner right",
        "y": "overlap",
      }
    }
    cascadingAnchor={
      Object {
        "x": "right",
        "y": "overlap",
      }
    }
    cascadingZDepth={3}
    className="header__menu-container"
    fixedTo={[Window]}
    id="header-menu"
    listHeightRestricted={true}
    listStyle={
      Object {
        "right": 10,
        "top": "100%",
      }
    }
    listZDepth={2}
    onClick={[Function]}
    onClose={[Function]}
    onKeyPress={[Function]}
    position="tr"
    repositionOnResize={false}
    repositionOnScroll={true}
    sameWidth={false}
    simplified={true}
    toggle={
      <AccessibleFakeButton
        className="header__menu-button"
        component="div"
        listenToEnter={true}
        listenToSpace={true}
        noFocusOutline={true}
        role="button"
        tabIndex={0}
      />
    }
    visible={false}
  />
</header>
`;
